### PR TITLE
feat: add forFeature configuration

### PR DESCRIPTION
**Pull request by va-stefanek**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #104 

## What is the new behavior?

Allow to setup hot-toast for dedicated with dedicated configuration

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngneat/hot-toast/src/lib/hot-toast.module.ts [hunk 1]
```diff
@@ -31,7 +32,11 @@ export class HotToastModule {
   static forRoot(config?: Partial<ToastConfig>): ModuleWithProviders<HotToastModule> {
     return {
       ngModule: HotToastModule,
-      providers: [{ provide: ToastConfig, useValue: config }],
+      providers: [{ provide: ToastConfig, useValue: config }, HotToastService],
```
Comment (shhdharmen) [inline projects/ngneat/hot-toast/src/lib/hot-toast.module.ts:35]: HotToastService is already provided in root.

**Review by shhdharmen — CHANGES_REQUESTED** [review]

File: projects/ngneat/hot-toast/src/lib/hot-toast.module.ts (hunk 1 above)
Comment (va-stefanek) [reply projects/ngneat/hot-toast/src/lib/hot-toast.module.ts:35]: Yea I will change it, that  there will be possibly multiple instances of HotToastService. I revert it by mistake sorry

**Comment by shhdharmen** [comment]
Please add relevant information in README and also add test cases.

**Review by shhdharmen — CHANGES_REQUESTED** [review]
Please add relevant information in README and also add test cases.

**Review by shhdharmen — COMMENTED** [review]
Add test cases

**Comment by shhdharmen** [comment]
@va-stefanek yeah, add a feature module in demo app and test it with Cypress.

File: src/app/app.module.ts [hunk 2]
```diff
@@ -32,7 +45,7 @@ import { ReverseOrderComponent } from './sections/reverse-order/reverse-order.co
     InjectorComponent,
     DataComponent,
   ],
-  imports: [BrowserModule, BrowserAnimationsModule, SharedModule, HotToastModule.forRoot({})],
+  imports: [BrowserModule, BrowserAnimationsModule, SharedModule, HotToastModule.forRoot({}), RouterModule, RouterModule.forRoot(routes)],
```
Comment (shhdharmen) [inline src/app/app.module.ts:48]: Imported `RouterModule` twice.

**Review by shhdharmen — CHANGES_REQUESTED** [review]

**Review by shhdharmen — APPROVED** [review]

**Comment by va-stefanek** [comment]
@shhdharmen done

**Comment by shhdharmen** [comment]
@va-stefanek `forFeature` is not working as expected for non-lazy loaded modules.

**Pull request merged**